src/app.py

```diff
@@ -4212,6 +4213,172 @@ def multiple_components():
     return jsonify(normalized_complete_entity_list)
 
 
+# Bulk update the entities in the entity-api.
+#
+# This function supports request throttling and retries.
+#
+# Parameters
+# ----------
+# entity_updates : dict
+#     The dictionary of entity updates. The key is the uuid and the value is the
+#     update dictionary.
+# token : str
+#     The groups token for the request.
+# entity_api_url : str
+#     The url of the entity-api.
+# total_tries : int, optional
+#     The number of total requests to be made for each update, by default 3.
+# throttle : float, optional
+#     The time to wait between requests and retries, by default 5.
+# after_each_callback : Callable[[int], None], optional
+#     A callback function to be called after each update, by default None. The index
+#     of the update is passed as a parameter to the callback.
+#
+# Returns
+# -------
+# dict
+#     The results of the bulk update. The key is the uuid of the entity. If
+#     successful, the value is a dictionary with "success" as True and "data" as the
+#     entity data. If failed, the value is a dictionary with "success" as False and
+#     "data" as the error message.
+def bulk_update_entities(
+    entity_updates: dict,
+    token: str,
+    entity_api_url: str,
+    total_tries: int = 3,
+    throttle: float = 5,
+    after_each_callback: Optional[Callable[[int], None]] = None,
+) -> dict:
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "X-Application": SchemaConstants.ENTITY_API_APP,
```

Comment:
@ChuckKollar I tested on DEV, and noticed this error:
```
[pid: 238|app: 0|req: 1198/2198] 127.0.0.1 () {40 vars in 690 bytes} [Tue Aug  6 21:35:43 2024] PUT /entities/4b2bf7a41109a010189bd9daa036500d => generated 103 bytes in 159 msecs (HTTP/1.1 400) 2 headers in 81 bytes (1 switches on core 1)
[2024-08-06 21:35:43] DEBUG in connectionpool: http://localhost:8080 "PUT /entities/4b2bf7a41109a010189bd9daa036500d HTTP/1.1" 400 103
[2024-08-06 21:35:43] INFO in app: bulk_update_entities() results: {'da3ec3e82a151f25e90aaf4631d5438e': {'success': False, 'data': '400 Bad Request: Unable to proceed due to missing X-Hubmap-Application header from request'}, '4b2bf7a41109a010189bd9daa036500d': {'success': False, 'data': '400 Bad Request: Unable to proceed due to missing X-Hubmap-Application header from request'}}
[2024-08-06 21:35:43] ERROR in app: Failed to update entity 4b2bf7a41109a010189bd9daa036500d: 400 Bad Request: Unable to proceed due to missing X-Hubmap-Application header from request
```

We'll just need to use `X-Hubmap-Application` instead of `X-Application` header. Please fix and make a new PR into `dev-integrate` branch.